addons/vscode/src/extension.ts

```diff
@@ -81,17 +82,17 @@ export async function getCliPath(extensionPath?: string): Promise<string> {
 	state.configPath = configPath;
 
 	const executableExists = (path: string) => {
-		return new Promise(resolve => {
-			try {
-				const spawnRet = spawn(path, ['--help'], {
-					timeout: 1000, /// 1 second
-				});
-				spawnRet.on('error', () => resolve(false));
-				spawnRet.on('exit', (code: number) => resolve(code === 0));
-			} catch {
-				resolve(false);
-			}
-		});
+		const child = spawnSync(path, ['-V'], { timeout: 1000, encoding: 'utf8' });
+		if (child.error) {
+			throw child.error;
+		}
+		if (child.status !== 0) {
+			throw child.status;
+		}
+		if (child.stdout.trim() !== `${name} ${version}`) {
+			throw Error(`version mismatch, expected ${name} ${version}, got ${child.stdout}`);
```

Comment:
I think we can raise a soft warning here.

Reply:
it should be a hard one IMO. We should always ensure js and rust are of the same version. Otherwise the behavior is undefined

Reply:
I think that means two philosophy, both are OK.
1. the one is people will continue their undefined behavior and report that they are getting warning if they want to report or the undefined behavior is really making program dead.
2. the other one is people will never get it work and find a way to ensure js and rust are of the same version.